### PR TITLE
MTEV_MAYBE implementation.

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/utils/mtev_maybe_alloc.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  type *name = __##name##_support.static_buff
+#define MTEV_MAYBE_SIZE(name) (__##name##_support.sz)
+#define MTEV_MAYBE_REALLOC(name, cnt) \
+do { \
+  if(__##name##_support.sz < (cnt) * sizeof(*(name))) { \
+    __##name##_support.sz = (cnt) * sizeof(*(name)); \
+    if(name != __##name##_support.static_buff) { \
+      name = realloc(name, __##name##_support.sz); \
+    } else { \
+      name = malloc(__##name##_support.sz); \
```
Comment (aidancully) [inline src/utils/mtev_maybe_alloc.h]: should this also do a `memcpy`?

**Comment by rileyberton** [comment]
Looks good to me

File: src/utils/mtev_maybe_alloc.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+#ifndef MTEV_MAYBE_ALLOC_H
+#define MTEV_MAYBE_ALLOC_H
+
+#include <stdlib.h>
+
+#define MTEV_DECL_MAYBE_ALLOC_VARS(type, name, cnt) \
+  struct { \
+    size_t sz; \
+    type   static_buff[cnt * sizeof(type)]; \
```
Comment (aidancully) [inline src/utils/mtev_maybe_alloc.h]: not `cnt * sizeof(type)`, just `cnt`

File: src/utils/mtev_maybe_alloc.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  } __##name##_support; \
+  type *name
+#define MTEV_INIT_MAYBE_ALLOC_VARS(type, name, cnt) \
+  __##name##_support.sz = cnt; \
```
Comment (aidancully) [inline src/utils/mtev_maybe_alloc.h]: `sz = cnt * sizeof(type);`

File: src/utils/mtev_maybe_alloc.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  name = __##name##__support.static_buff
+#define MTEV_DECL_MAYBE_ALLOC(type, name, cnt) \
+  struct { \
+    size_t sz; \
+    type   static_buff[cnt * sizeof(type)]; \
```
Comment (aidancully) [inline src/utils/mtev_maybe_alloc.h]: `static_buff[cnt]`

File: src/utils/mtev_maybe_alloc.h [hunk 5]
```diff
@@ -33,22 +33,57 @@
...
+/*! \fn MTEV_DECL_MAYBE_ALLOC_VARS(type, name, cnt)
+    \brief C Macro for declaring a "maybe" buffer.
+    \param type A C type (e.g. char)
+    \param name The name of the C variable to declare.
+    \param cnt The number of type elements initially declared.
+ */
 #define MTEV_DECL_MAYBE_ALLOC_VARS(type, name, cnt) \
   struct { \
     size_t sz; \
     type   static_buff[cnt * sizeof(type)]; \
```
Comment (aidancully) [inline src/utils/mtev_maybe_alloc.h]: `static_buff[cnt]`

**Comment by aidancully** [comment]
:+1:

**Pull request merged**